internal/app/api/v1/scripts.go

```diff
@@ -225,20 +232,61 @@ func (s kubtestAPI) ListExecutions() fiber.Handler {
 			return s.Error(c, http.StatusInternalServerError, err)
 		}
 
-		// convert to summary
-		result := make([]kubtest.ExecutionSummary, len(executions))
-		for i, s := range executions {
-			result[i] = kubtest.ExecutionSummary{
-				Id:         s.Id,
-				ScriptName: s.ScriptName,
-				ScriptType: s.ScriptType,
-				Status:     s.Execution.Status,
-				StartTime:  s.Execution.StartTime,
-				EndTime:    s.Execution.EndTime,
+		totals := kubtest.ExecutionTotals{
+			Results: int32(len(executions)),
+			Passed:  0,
+			Failed:  0,
+			Queued:  0,
+			Pending: 0,
+		}
+
+		executionResults := make([]kubtest.ExecutionSummary, pageSize)
+		addedToResultCount := 0
+		filteredCount := 0
+
+		for _, s := range executions {
+
+			switch *s.Execution.Status {
+			case kubtest.QUEUED_ExecutionStatus:
+				totals.Queued++
+				break
+			case kubtest.SUCCESS_ExecutionStatus:
+				totals.Passed++
+				break
+			case kubtest.FAILED_ExecutionStatus:
+				totals.Failed++
+				break
+			case kubtest.PENDING_ExecutionStatus:
+				totals.Pending++
+				break
 			}
+
+			if addedToResultCount < pageSize {
+				if statusFilter == "" || string(*s.Execution.Status) == statusFilter {
```

Comment:
These 2 conditions can be merged together in order to keep the nesting level as low as possible.

Reply:
I would extract method from this calculation logic - handler starts to be fat :D 
can be part of Executions type - `Totals()` method - would be easier to test :)